gslib/context_config.py

```diff
@@ -124,37 +186,29 @@ def __init__(self, logger):
     self.use_client_certificate = config.getbool('Credentials',
                                                  'use_client_certificate')
     self.client_cert_path = None
-    self.client_cert_password = None
-
     if not self.use_client_certificate:
       # Don't spend time generating values gsutil won't need.
       return
 
     # Generates certificate and deletes it afterwards.
     atexit.register(self._UnprovisionClientCert)
-    command_string = config.get('Credentials', 'cert_provider_command', None)
-    if not command_string:
-      raise CertProvisionError('No cert provider detected.')
-
     self.client_cert_path = os.path.join(gslib.GSUTIL_DIR, 'caa_cert.pem')
     try:
       # Certs provisioned using endpoint verification are stored as a
       # single file holding both the public certificate and the private key.
-      self._ProvisionClientCert(command_string, self.client_cert_path)
+      self._ProvisionClientCert(self.client_cert_path)
     except CertProvisionError as e:
       self.logger.error('Failed to provision client certificate: %s' % e)
 
-  def _ProvisionClientCert(self, command_string, cert_path):
+  def _ProvisionClientCert(self, cert_path):
     """Executes certificate provider to obtain client certificate and keys."""
-    # Monkey-patch command line args to get password-protected certificate.
-    # Adds password flag if it's not already there.
-    password_arg = ' --with_passphrase'
-    if ('--print_certificate' in command_string and
-        password_arg not in command_string):
-      command_string += password_arg
+    cert_command = config.get('Credentials', 'cert_provider_command', None)
+    if not cert_command:
+      # Load the default certificate provider if sit is not provided by user.
+      cert_command = _default_command()
 
     try:
-      command_process = subprocess.Popen(command_string.split(' '),
+      command_process = subprocess.Popen(cert_command,
```

Comment:
The kokoro failures are happening because we are no longer splitting the command here. Any reason why this was removed?

Reply:
Oh, I see that you cert command can be a list if it is not defined in the boto config file as per your change. But if it is defined in the config file, it will be a list. 
So you will have t omake sure that cert_command is of the same type, either a string or a list.

after line 205, you can add the split call.


For kokoro test, we are passing a cat <filename> command here https://github.com/GoogleCloudPlatform/gsutil/blob/5317bc2f19c2c6be8e0a502f209f369c39eb9d2f/test/ci/kokoro/config_generator.sh#L49

Reply:
You are right. I put the split(' ') back for the boto config command. Thanks.